core/src/main/scala/chisel3/reflect/DataMirror.scala

```diff
@@ -160,20 +160,21 @@ object DataMirror {
     *       of its children.
     * @see [[DataMirror.modulePorts]] for a non-recursive representation of the ports.
     */
-  def fullModulePorts(target: BaseModule)(implicit si: SourceInfo): Seq[(String, Data)] = {
-    def getPortNames(name: String, data: Data): Seq[(String, Data)] = Seq(name -> data) ++ (data match {
-      case _: Element => Seq()
-      case r: Record =>
-        r._elements.toSeq.flatMap {
-          case (eltName, elt) =>
-            if (r._isOpaqueType) { getPortNames(s"${name}", elt) }
-            else { getPortNames(s"${name}_${eltName}", elt) }
-        }
-      case v: Vec[_] => v.zipWithIndex.flatMap { case (elt, index) => getPortNames(s"${name}_${index}", elt) }
-    })
+  def fullModulePorts(target: BaseModule)(implicit si: SourceInfo): Seq[(String, IsBindable)] = {
```

Comment:
I'm a little iffy on change the return type of these public methods, this is where this change might break user code. Can we try this out in a large codebase first?

Reply:
I believe all of the public method return types that were touched were here in DataMirror. If it's an issue, perhaps we can do the filtering down to just Data like some of the other places, since it is... DataMirror. I think it would be reasonable for DataMirror to not support non-Data types initially, but I need to double check if that would satisfy the initial use-cases I have in mind.

Reply:
Yeah, these ones will affect downstream users. If we want this to be a non-breaking change I think I can let this stuff in DataMirror just return Data as one option.

Reply:
Ok, I have updated DataMirror to continue returning Data from these public APIs in a local branch. I also found one other "public API"--the protected method `getModulePorts` on `BaseModule`, and filtered that one to return Data ports only as well.

With this, I was able to test on a large downstream codebase and confirm there were no compile time issues. I've also generated a large design and ran some basic simulations, and they pass.

Does this seem like a good plan? I am not the expert here, but it seems safer to keep the return types Data in these APIs initially, and we can generalize them if/when it becomes necessary. The non-Data types are all new, and I think it's reasonable for certain APIs to not return ports of those types initially.